### PR TITLE
pkcs11/_pkcs11.pyx: Add reinitalize func to lib class

### PR DESCRIPTION
**Pull request by grant100**
Per our conversation a while back I'm resubmitting this as a single function.

### CODE REVIEW

File: pkcs11/_pkcs11.pyx [hunk 1]
```diff
@@ -1355,6 +1355,11 @@ cdef class lib:
         except StopIteration:
             return token
 
+    def reinitialize(self):
```
Comment (danni) [inline pkcs11/_pkcs11.pyx:1358]: Can you please add a docstring? It will also need one in the *.py file that will be generated into the API docs.

Comment (danni) [reply pkcs11/_pkcs11.pyx:1358]: FYI docstrings go in `types.py`

Comment (grant100) [reply pkcs11/_pkcs11.pyx:1358]: Perhaps you can help me understand, but there is no class in types.py that the `cdef class Lib` shadows. How would you like me to proceed?

Comment (danni) [reply pkcs11/_pkcs11.pyx:1358]: Ah you're right, `lib` is magic and so wouldn't auto-inspect. It's manually documented in `api.rst` from line 15 onwards.

**Pull request merged**